src/client/Client.cc

```diff
@@ -9785,6 +9785,9 @@ int Client::statfs(const char *path, struct statvfs *stbuf,
 
   client_lock.Unlock();
   int rval = cond.wait();
+
+  unsigned long int total_files_on_fs = get_root()->rstat.rfiles +
```

Comment:
When you use get_root like this, you're still accessing the inode attributes outside of the lock, and get_root is also calling ll_get, leaving an unnecessary increment to the reference count.

Instead, just access `Client::root` directly while holding client_lock

Reply:
> When you use get_root like this, you're still accessing the inode attributes outside of the lock, and get_root is also calling ll_get, leaving an unnecessary increment to the reference count.

Not outside since `Client::get_root` itself [locks](https://github.com/rishabh-d-dave/ceph/blob/master/src/client/Client.cc#L363) the `client_lock`.  :)

> Instead, just access Client::root directly while holding client_lock. Include a check for root==nullptr in case statfs gets called very early, before we've loaded the root inode.

Done.

Reply:
> Not outside since Client::get_root itself locks the client_lock. :)

The code inside get_root was in the lock, but the use of the Inode after it returned was unlocked.